sdk/src/types/block/payload/mod.rs

```diff
@@ -74,6 +75,23 @@ impl Payload {
     crate::def_is_as_opt!(Payload: SignedTransaction, TaggedData);
 }
 
+impl WorkScore for Payload {
+    fn work_score(&self, work_score_params: WorkScoreParameters) -> u32 {
+        // 1 byte for the payload kind
+        let score = size_of::<u8>() as u32
```

Comment:
We should consider the implication of this, since currently calling `SignedTransactionPayload::work_score` for instance will not include the type byte.

Reply:
that's correct ... and thanks for pointing that out as we could include the type byte in the payload type itself. However, we do pack in the same way. Adding the type byte only happens for the packed `Payload`, not for the concrete payloads. But maybe @thibault-martinez knows whether for work scores we do it differently.